frontend/src/pages/NewAgreement.tsx

```diff
@@ -1,43 +1,68 @@
-import { gql, useMutation } from '@apollo/client';
+import { useQuery, useLazyQuery, useMutation } from '@apollo/client';
 import { useTranslation } from 'react-i18next';
 import { StringBank } from '../strings';
 import { useTheme } from '@mui/material/styles';
 import React, { FC } from 'react';
 import { useState, useEffect, useContext, useCallback, useRef, useLayoutEffect } from 'react';
-import { CategorySelect } from '../components';
-import { Button, Stack, Typography, InputBase } from '@mui/material';
+import {
+  agreementDetailsQuery,
+  agreementTopicsAndSectionsQuery,
+  addAgreementMutation,
+  updateAgreementMutation,
+} from 'utils/queries';
+import { CategorySelect, NewTopicsEditor } from '../components';
+import { Button, Stack, Typography, InputBase, Divider } from '@mui/material';
 import EditOutlinedIcon from '@mui/icons-material/EditOutlined';
 import { GroupContext } from 'contexts/group';
 
-// Mutation to insert new agreement record
-const ADD_AGREEMENT = gql`
-  mutation AddAgreement($category_id: Int!, $group_id: Int!, $name: String!, $rationale: String!) {
-    insert_core_agreements_one(
-      object: { category_id: $category_id, group_id: $group_id, name: $name, rationale: $rationale }
-    ) {
-      id
-      category_id
-      group_id
-      name
-      rationale
-    }
-  }
-`;
-
 const NewAgreement: FC = () => {
   const { i18n, t } = useTranslation();
   const { id } = useContext(GroupContext);
   const theme = useTheme();
 
+  /**
+   * Saved agreement state (in case of browser refresh)
+   */
+  const sessionStorageKey = 'consenz-new-agreement-id';
+  const [agreementId, setAgreementId] = useState(
+    sessionStorage.getItem(sessionStorageKey) as unknown | null as number | null
+  );
+  const [savedData, setSavedData] = useState<{
+    id: number;
+    name: string;
+    category_id: number;
+    rationale: string;
+  } | null>(null);
+  const {
+    // loading: agreementQueryLoading,
+    error: agreementQueryError,
+    // data: agreementQueryData,
+  } = useQuery(agreementDetailsQuery, {
+    variables: { id: agreementId },
+    onCompleted: (data) => {
+      const record = data.core_agreements?.at(0);
+      if (!record || !record.name || !record.rationale) return;
+      setSavedData(record);
+      setAgreementName(record.name);
+      setIsNameEdited(true);
+      setCategoryId(record.category_id);
+      setRationale(record.rationale);
+      setIsRationaleEdited(true);
+      setTopicsAndSections(record.topics as Array<object>);
+    },
+    skip: agreementId === null || savedData !== null,
+  });
+  if (agreementQueryError) console.log(agreementQueryError);
+
```

Comment:
This seems to be redundant. Am I missing something here?

Reply:
This query exclusively serves the case of browser refresh during step 2, which is why it is skipped under any other scenario.

Reply:
Oh I see :) 
So let's just clean this up a little bit and remove the lines which are commented out